### ISSUE TITLE
Deployment on Mac

### ISSUE THREAD

**Issue by wveinti** [dotnet-architecture/eShopOnContainers#1124]
Hi 

I'm following all the steps from the guide to deploy the project on Mac Os (Mojave).  I don't really get an error but when I load  http://docker.for.mac.localhost:5105/  , I get the eshoponcontainers site but it shows  a message (You can also see the screenshot)
If you could please provide me with any idea what I need to change or fix to get the site up and running?

**Welcome to IdentityServer4**
IdentityServer publishes a discovery document where you can find metadata and links to all the endpoints, key material, etc.

Here are links to the source code repository, and ready to use samples.
<img width="1199" alt="Screen Shot 2019-08-20 at 2 08 09 PM" src="https://user-images.githubusercontent.com/17145603/63372916-0f6dc780-c355-11e9-97c1-6239faba5897.png">

**Comment by ognjen-blazic** [comment]
Well, as stated in `docker-compose.override.yml`, IS4 is running on that endpoint (`IdentityUrl=http://docker.for.mac.localhost:5105`) in docker container.
So, it is not an error, you just need to browse to other address, either client one or login page that is served on IS4.

**Comment by wveinti** [comment]
Thank you so much for your help.
I went to this address and it is working
http://docker.for.mac.localhost:5100

<img width="1208" alt="Screen Shot 2019-08-23 at 11 16 09 AM" src="https://user-images.githubusercontent.com/17145603/63603401-9ca05e80-c597-11e9-93ff-3055f2868f8c.png">

But when I hit the login button I get an error, Please advise.. thanks!
Error
Sorry, there was an error : unauthorized_client
Request Id: 0HLP6EI615BC9:00000001

<img width="1039" alt="Screen Shot 2019-08-23 at 11 12 43 AM" src="https://user-images.githubusercontent.com/17145603/63603457-b772d300-c597-11e9-8eb1-821b85cceef3.png">

**Comment by ognjen-blazic** [comment]
You would need to dig deep into how IS4 is working and see [similar](https://stackoverflow.com/questions/52655664/identity-server-4-why-i-receive-unauthorized-client/52659346) _StackOverflow_ questions.

But, from my previous knowledge, inspect what request URL is sent to IS, it contains `client_id` and other important parameters. They must match IS client configuration which is probably in `Config.cs` file in IS project itself.

In other words, IS is not recognizing your client application that is trying to login.

**Comment by mvelosop** [comment]
Hi @wveinti I'm sorry for the delayed response.

Just yesterday I got my hands in a Mac to run eShopOnContainers and everything ran much smoother than I expected. Specially considering this was my first time ever with a Mac 😅.

BTW Thanks @ognjen-blazic.

Anyway, the problem with the login is caused by the initial configuration in the `.env` file (fixed that in the wiki for the upcoming .NET Core 3.0 migration).

You have to update the `.env` file as follows:

```yaml
ESHOP_EXTERNAL_DNS_NAME_OR_IP=docker.for.mac.localhost
```

The issue here is that when starting up for the first time, the **IdentityServer** database is seeded with the allowed clients, taking the information from the environment variables below, in `docker-compose.override.yml`.

```yaml
identity-api:
  environment:
    - ASPNETCORE_ENVIRONMENT=Development
    - ASPNETCORE_URLS=http://0.0.0.0:80
    - SpaClient=http://${ESHOP_EXTERNAL_DNS_NAME_OR_IP}:5104
    - XamarinCallback=http://${ESHOP_PROD_EXTERNAL_DNS_NAME_OR_IP}:5105/xamarincallback
    - ConnectionString=${ESHOP_AZURE_IDENTITY_DB:-Server=sqldata;Database=Microsoft.eShopOnContainers.Service.IdentityDb;User Id=sa;Password=Pass@word}
    - MvcClient=http://${ESHOP_EXTERNAL_DNS_NAME_OR_IP}:5100
    - LocationApiClient=http://${ESHOP_EXTERNAL_DNS_NAME_OR_IP}:5109
    - MarketingApiClient=http://${ESHOP_EXTERNAL_DNS_NAME_OR_IP}:5110
    - BasketApiClient=http://${ESHOP_EXTERNAL_DNS_NAME_OR_IP}:5103
    - OrderingApiClient=http://${ESHOP_EXTERNAL_DNS_NAME_OR_IP}:5102
    - MobileShoppingAggClient=http://${ESHOP_EXTERNAL_DNS_NAME_OR_IP}:5120
    - WebShoppingAggClient=http://${ESHOP_EXTERNAL_DNS_NAME_OR_IP}:5121
    - WebhooksApiClient=http://${ESHOP_EXTERNAL_DNS_NAME_OR_IP}:5113
    - WebhooksWebClient=http://${ESHOP_EXTERNAL_DNS_NAME_OR_IP}:5114
    - UseCustomizationData=True
    - ApplicationInsights__InstrumentationKey=${INSTRUMENTATION_KEY}
    - OrchestratorType=${ORCHESTRATOR_TYPE}
  ports:
    - "5105:80"
```

So with the initial configuration, IdentityServer is left configured to accept connections from `localhost`, not from `docker.for.mac.localhost`.

The simplest solution is to drop the `sqldata` container so, when starting up again, the DB is seeded with the `docker.for.mac.localhost` clients.

To achieve the above, just run tese commands:

```console
docker stop $(docker ps -aqf name=sqldata)
docker rm -v $(docker ps -aqf name=sqldata)
```

Then, after updating the `.env` file as shown above. just start again with `docker-compose up`

Hope this helps.

**Comment by mvelosop** [comment]
Closing this issue now but feel free to comment, will reopen if needed.

**Comment by adahhane** [comment]
Hello @mvelosop 
I tried you suggestion but it doesn't work for me.
I follow step by step with no results.
There is my configuration,
![Capture d’écran 2020-09-15 à 15 47 09](https://user-images.githubusercontent.com/656154/93218767-df1c7680-f76a-11ea-8ed3-f3096d1089dc.png)
![Capture d’écran 2020-09-15 à 15 46 18](https://user-images.githubusercontent.com/656154/93218772-e04da380-f76a-11ea-8c8e-73cce717c2bd.png)
![Capture d’écran 2020-09-15 à 15 46 06](https://user-images.githubusercontent.com/656154/93218774-e04da380-f76a-11ea-89af-0e66b60f513a.png)
![Capture d’écran 2020-09-15 à 15 45 52](https://user-images.githubusercontent.com/656154/93218779-e2affd80-f76a-11ea-967e-6577f963fbae.png)

Maybe I missed up something
Thanks in advance
